lib/codegen/src/isa/x86/enc_tables.rs

```diff
@@ -5,15 +5,73 @@ use bitset::BitSet;
 use cursor::{Cursor, FuncCursor};
 use flowgraph::ControlFlowGraph;
 use ir::condcodes::IntCC;
-use ir::{self, InstBuilder};
+use ir::{self, Function, Inst, InstBuilder};
 use isa;
 use isa::constraints::*;
 use isa::enc_tables::*;
+use isa::encoding::base_size;
 use isa::encoding::RecipeSizing;
+use isa::RegUnit;
+use regalloc::RegDiversions;
 
 include!(concat!(env!("OUT_DIR"), "/encoding-x86.rs"));
 include!(concat!(env!("OUT_DIR"), "/legalize-x86.rs"));
 
+pub fn needs_sib_byte(reg: RegUnit) -> bool {
+    reg == RU::r12 as u16 || reg == RU::rsp as u16
+}
+pub fn needs_offset(reg: RegUnit) -> bool {
+    reg == RU::r13 as u16 || reg == RU::rbp as u16
+}
+
+fn _additional_size_if(
```

Comment:
This function is unconditionally used below; can we remove the leading underscore?